register/register.go

```diff
@@ -0,0 +1,162 @@
+package register
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/gin-gonic/gin"
+	"github.com/gin-gonic/gin/binding"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/jmoiron/sqlx"
+)
+
+type UserRegister struct {
+	Name     string `json:"name"`
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+type RegisterRepository struct {
+	Db *sqlx.DB
+}
+type Claims struct {
+	Username string `json:"username"`
+	jwt.StandardClaims
+}
+
+// Create the JWT key used to create the signature
+var jwtKey = []byte("my_secret_key")
+
+func New(db *sqlx.DB) *RegisterRepository {
+	return &RegisterRepository{Db: db}
+}
+
+func (repository *RegisterRepository) AddUser(c *gin.Context) {
+
+	input := UserRegister{}
+	err := c.ShouldBindWith(&input, binding.JSON)
+
+	if err != nil {
+		c.Abort()
+		c.JSON(http.StatusBadRequest, gin.H{"message cannot bind the STRUCT ": err.Error()})
+		return
+	}
+
+	_, err = repository.Db.Exec(`INSERT INTO Users(name,email,password,user_type) VALUES (?,?,?,?)`, input.Name, input.Email, input.Password, "General")
+
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"message cannot insert ": err.Error()})
```

Comment:
We have to throw errors based on error scenarios. Every time we can not throw `badRequest`.
If it is a `server error` then  `InternalServerError`
If `user is not authorized` then `Unauthorized` error